crates/store/src/db/models/queries/nullifiers.rs

```diff
@@ -50,17 +66,37 @@ pub(crate) fn select_nullifiers_by_prefix(
     // WHERE
     //     nullifier_prefix IN rarray(?1) AND
     //     block_num >= ?2
+    //     AND block_num <= ?3
     // ORDER BY
     //     block_num ASC
+    // LIMIT
+    //     ROW_LIMIT;
 
     let prefixes = nullifier_prefixes.iter().map(|prefix| nullifier_prefix_to_raw_sql(*prefix));
-    let nullifiers_raw =
+    let raw =
         SelectDsl::select(schema::nullifiers::table, NullifierWithoutPrefixRawRow::as_select())
             .filter(schema::nullifiers::nullifier_prefix.eq_any(prefixes))
-            .filter(schema::nullifiers::block_num.ge(block_num.to_raw_sql()))
+            .filter(schema::nullifiers::block_num.ge(block_from.to_raw_sql()))
+            .filter(schema::nullifiers::block_num.le(block_to.to_raw_sql()))
             .order(schema::nullifiers::block_num.asc())
+            .limit(i64::try_from(ROW_LIMIT).expect("limit fits within i64"))
             .load::<NullifierWithoutPrefixRawRow>(conn)?;
-    vec_raw_try_into(nullifiers_raw)
+
+    // Discard the last block in the response (assumes more than one block may be present)
+    if raw.len() >= ROW_LIMIT {
```

Comment:
nit: With newer rust versions we should be able to do `if let Some(last) = raw.last() && raw.len() >= ROW_LIMIT {`